sympy/core/mul.py

```diff
@@ -1255,19 +1255,20 @@ def _eval_is_integer(self):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
-            if d is S.One:
+            if d == S.One:
                 return True
             elif d == S(2):
                 return n.is_even
-        # if d is even -- 0 or not -- the
-        # result is not an integer
         if n.is_odd and d.is_even:
             return False
+        if n == d:
```

Comment:
`1.0 == 1` but 1.0 is not an integer to SymPy

Reply:
I just added that to add support for cases like these
```python3
>>> fraction(Mul( 5, 1/5, evaluate=False))
(1.00000000000000, 1)
>>> fraction(Mul( 5, Rational(1, 5), evaluate=False))
(5, 5)
```

---

```python3
>>> Mul( 5, 1/5, evaluate=False).is_integer
True
>>> Mul( 5, Rational(1,5), evaluate=False).is_integer
True
```

Reply:
> I just added that to add support for cases like these

That's fine but you have to address the case for when it's not fine: `if n == d: return fuzzy_and(n.is_integer, d.is_integer)`. So `Mul(2., S.Half, evaluate=False).is_integer -> False`

I guess there would also be the case:
```
if ((n - 2).is_zero or (n + 2).is_zero):
    return (d - 2).is_positive
```
but that would require fixing Add._eval_is_integer to look for args that are prime and even.

Reply:
> So `Mul(2., S.Half, evaluate=False).is_integer -> False`

shouldn't this be true ?

Reply:
No because `S(1.).is_integer -> False`

Reply:
Ohhh okay!, I just assumed it was an integer cause floor(1.) == S.One, like the condition we used in Binomial Piecewise